sdk/src/wallet/utxo.ts

```diff
@@ -69,20 +70,28 @@ export async function createBitcoinPsbt(
     const addressInfo = getAddressInfo(fromAddress);
 
     // TODO: possibly, allow other strategies to be passed to this function
-    const utxoSelectionStrategy = 'default';
+    let utxoSelectionStrategy = 'default';
+    if (addressInfo.type === AddressType.p2tr) {
+        utxoSelectionStrategy = 'all';
+    }
 
     if (addressInfo.network === 'regtest') {
         throw new Error('Bitcoin regtest not supported');
     }
 
     // We need the public key to generate the redeem and witness script to spend the scripts
-    if (addressInfo.type === (AddressType.p2sh || AddressType.p2wsh)) {
+    if (
+        addressInfo.type === AddressType.p2sh ||
+        addressInfo.type === AddressType.p2wsh ||
+        addressInfo.type === AddressType.p2tr
+    ) {
```

Comment:
Oh, you are right. We still need the public key to spend a p2wsh though since we can use the public key to generate the witness script that we need to spend the p2wsh (assuming this is a p2wsh-p2pkh or similar). If it was a multisig or other things under the hood, we'd need more info in the witness script generation.